.github/workflows/test-unit.yaml

```
@@ -0,0 +1,32 @@
+name: test-unit
+
+on:
+  - pull_request
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@v3.0.1
+
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+
+      - name: Run Tests
+        run: |
+          make test-unit xml=true
```

Maybe we should always use html to report coverage or it is required for GitHub actions ?
```suggestion
          make test-unit
```